### PR TITLE
Feat(eos_designs): Add channel_id for endpoints

### LINKED ISSUE #2069: Allow channel id to be specified for port channels to endpoints
### Enhancement summary

The current behaviour for Port Channel for Endpoints is that the number of the first interface is used to generate the 
channel id.
This proposal is to support an optional `channel_id` where the specific id can be supplied by the user, similar to how it can be specified for the mlag port channel with `mlag_port_channel_id`.

### Which component of AVD is impacted

eos_designs

### Use case example

In the below example, we need the port channel to have an id of 211 instead of 11  which is the default that AVD would generate 

```yaml
servers:
  Server1:
    rack: Rack1
    adapters:
      - endpoint_ports: [E0, E1]
        switch_ports: [Ethernet1/1, Ethernet1/1]
        switches: [SW-1, SW-2]
        profile: PROFILE_1
        port_channel:
          channel_id: 211
          description: myportchannel
          mode: active
```

The structured_configuration that is generated by eos_designs should have: 

```yml
interface Port-Channel211
   description Server1_myportchannel
   no shutdown
   switchport
   switchport trunk allowed vlan 210,220
   switchport trunk native vlan 220
   switchport mode trunk
   mlag 211
   spanning-tree portfast
!
```

### Describe the solution you would like

Ideally the builtin templates of AVD would support this optional value.

### Describe alternatives you have considered

At the moment we are achieving this using custom jinja templates to override the default AVD eos_designs templates for port-channel configuration. However, with updates to AVD, these templates would not include any other improvements made to the built-in templates. 

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by chetryan**
## Change Summary

Add the ability to specify the channel_id to be used for port-channels for endpoints. If the key is missing, the existing default behaviour applies, where the id is created from the interface number of the first switch port in the port-channel. 

## Related Issue(s)

Fixes #2069 

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

Add a new key `channel_id` to the port channel of an endpoint definition, as per the below example:

```yaml
server:
  server22_port_channel_with_custom_id:
    rack: RackC
    adapters:
      - server_ports: [ Eth1, Eth2 ]
        switch_ports: [ Ethernet29, Ethernet29 ]
        switches: [ DC1-SVC3A, DC1-SVC3B ]
        profile: TENANT_A
        port_channel:
          description: server22_port_channel_with_custom_id
          mode: active
          enabled: true
          channel_id: 1291
```
This is achieve by updating the `ansible_collections/arista/avd/roles/eos_designs/templates/connected_endpoints/port-channel-interfaces.j2` and 
`ansible_collections/arista/avd/roles/eos_designs/templates/connected_endpoints/ethernet-interfaces.j2`
 by adding the following logic to both: 

```jinja
{%                 if adapter_settings.port_channel.channel_id is arista.avd.defined %}
{%                     set channel_group_id = adapter_settings.port_channel.channel_id %}
{%                 else %}
{%                     set channel_group_id = (adapter_settings.switch_ports[0] | regex_findall("\d") | join) %}
{%                 endif %}
```

## How to test

molecule test --scenario-name eos_designs_unit_tests --

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Review by carlbuchmann — APPROVED** [review]
LGTM!

File: ansible_collections/arista/avd/roles/eos_designs/templates/connected_endpoints/ethernet-interfaces.j2 [hunk 1]
```diff
@@ -29,7 +29,11 @@ ethernet_interfaces:
 {%                         do monitor_sessions_data.append(monitor_session) %}
 {%                     endfor %}
 {%                 endif %}
-{%                 set channel_group_id = (adapter_settings.switch_ports[0] | regex_findall("\d") | join) %}
+{%                 if adapter_settings.port_channel.channel_id is arista.avd.defined %}
+{%                     set channel_group_id = adapter_settings.port_channel.channel_id %}
+{%                 else %}
+{%                     set channel_group_id = (adapter_settings.switch_ports[0] | regex_findall("\d") | join) %}
+{%                 endif %}
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/templates/connected_endpoints/ethernet-interfaces.j2]: Can use default here:

```suggestion
+{%                 set channel_group_id = adapter_settings.port_channel.channel_id | arista.avd.default(adapter_settings.switch_ports[0] | regex_findall("\d") | join) %}

```

File: ansible_collections/arista/avd/roles/eos_designs/templates/connected_endpoints/port-channel-interfaces.j2 [hunk 2]
```diff
@@ -18,7 +18,11 @@ port_channel_interfaces:
                                                adapter_profile | arista.avd.default({}),
                                                adapter, recursive=true, list_merge='replace') %}
 {%         if adapter_settings.port_channel.mode is arista.avd.defined %}
-{%             set channel_group_id = (adapter_settings.switch_ports[0] | regex_findall("\d") | join) %}
+{%             if adapter_settings.port_channel.channel_id is arista.avd.defined %}
+{%                 set channel_group_id = adapter_settings.port_channel.channel_id %}
+{%             else %}
+{%                 set channel_group_id = (adapter_settings.switch_ports[0] | regex_findall("\d") | join) %}
+{%             endif %}
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/templates/connected_endpoints/port-channel-interfaces.j2]:
```suggestion
+{%             set channel_group_id = adapter_settings.port_channel.channel_id | arista.avd.default(adapter_settings.switch_ports[0] | regex_findall("\d") | join) %}

```

**Review by gmuloc — COMMENTED** [review]
added some small refactoring suggestion leveraging `arista.avd.default`

File: ansible_collections/arista/avd/roles/eos_designs/templates/connected_endpoints/ethernet-interfaces.j2 (hunk 1 above)
Comment (carlbuchmann) [reply ansible_collections/arista/avd/roles/eos_designs/templates/connected_endpoints/ethernet-interfaces.j2]: Great suggestion @gmuloc!

File: ansible_collections/arista/avd/roles/eos_designs/templates/connected_endpoints/port-channel-interfaces.j2 (hunk 2 above)
Comment (carlbuchmann) [reply ansible_collections/arista/avd/roles/eos_designs/templates/connected_endpoints/port-channel-interfaces.j2]: Great suggestion @gmuloc!

**Comment by chetryan** [comment]
Thank you @gmuloc , was not aware we could use filters within `arista.avd.default`. That's really handy.
Fixed typo (extra + ) in the templates.

**Review by tgodaA — APPROVED** [review]
LGTM

**Pull request merged**